### ISSUE TITLE
generate-yaml failed, but out folder was generated

### ISSUE THREAD

**Issue by gyliu513** [kubernetes-sigs/cluster-api-provider-ibmcloud#253]
/kind bug
/sig ibmcloud

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

I was using kubectl v1.11.0 and kustomize failed due to we request kubectl 1.14.0+, but after generate-yaml.sh failed, there are still some files generated under out folder.

```
root@gyliu-dev1:~/go/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud# ./generate-yaml.sh ./clouds.yaml ubuntu
no output folder provided, use name 'out' by default
Error: unknown command "kustomize" for "kubectl"
Run 'kubectl --help' for usage.
unknown command "kustomize" for "kubectl"
```
```
root@gyliu-dev1:~/go/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud# kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T20:08:34Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1+icp-ee", GitCommit:"5803c3b1f9422c43a963e0610b3a4cad565e127e", GitTreeState:"clean", BuildDate:"2018-09-04T09:29:02Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```
```
root@gyliu-dev1:~/go/src/sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/clusterctl/examples/ibmcloud# ls out
cluster.yaml  machines.yaml  provider-components.yaml
```

**What did you expect to happen:**

I think we should not generate files if kubectl version is not right.

**Comment by jichenjc** [comment]
right, should we check kubectl version or clean up those files or both?

**Comment by gyliu513** [comment]
Yes, we should check kubectl version first, if kubectl version is not right, fail and return.

**Comment by gyliu513** [comment]
@wangtjia

**Comment by wangtjia** [comment]
The issue is fixed and passed test on local environment #255
